### ISSUE TITLE
Unknown issue with com.alee.utils or SearchWindow

### ISSUE THREAD

**Issue by mokun** [mars-sim/mars-sim#368]
**Describe the bug**
- I've been getting some unknown issues with com.alee.utils or with SearchWindow during start-up randomly.
- There is a few variations as to what trigger the crash and it seems to surround with one of the com.alee.utils.

**Expected behavior**
I never started the SearchWindow. Rather, it's around the time when I was going to open the Mission Tool and this exception pops up in the stacktrace.

**Desktop**
 - Build 5850

**Additional context**
```
00-Adir-01:010.021 (Info) LivingAccommodations : [New Plymouth - Lander Hab 1] Dawn Martinez - Designated a bed at (-3.04, -1.04).
00-Adir-01:010.021 (Info) LivingAccommodations : [New Plymouth - Residential Quarters 1] Lori Thomas - Designated a bed at (-9.8, 16.2).
00-Adir-01:010.021 (Info) LivingAccommodations : [New Plymouth - Residential Quarters 1] Kevin Bell - Designated a bed at (-9.8, 14.8).
00-Adir-01:010.021 (Info) LivingAccommodations : [New Plymouth - Residential Quarters 1] Frank Bell - Designated a bed at (-9.8, 13.4).
00-Adir-01:010.021 (Info) LivingAccommodations : [New Plymouth - Residential Quarters 1] Frances Davis - Designated a bed at (-9.8, 12.4).
[unitmanager-thread-0] ERROR com.alee.utils.ProprietaryUtils - java.lang.reflect.InvocationTargetException
java.lang.reflect.InvocationTargetException
	at jdk.internal.reflect.GeneratedMethodAccessor26.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at com.alee.utils.ReflectUtils.callMethod(ReflectUtils.java:1635)
	at com.alee.utils.SwingUtils.firePropertyChanged(SwingUtils.java:2944)
	at com.alee.painter.decoration.DecorationUtils.fireStatesChanged(DecorationUtils.java:166)
	at com.alee.laf.list.WebListCellRenderer.getListCellRendererComponent(WebListCellRenderer.java:309)
	at java.desktop/javax.swing.plaf.basic.BasicListUI.updateLayoutState(BasicListUI.java:1439)
	at java.desktop/javax.swing.plaf.basic.BasicListUI.maybeUpdateLayoutState(BasicListUI.java:1389)
	at java.desktop/javax.swing.plaf.basic.BasicListUI$Handler.valueChanged(BasicListUI.java:2785)
	at java.desktop/javax.swing.DefaultListSelectionModel.fireValueChanged(DefaultListSelectionModel.java:219)
	at java.desktop/javax.swing.DefaultListSelectionModel.fireValueChanged(DefaultListSelectionModel.java:199)
	at java.desktop/javax.swing.DefaultListSelectionModel.fireValueChanged(DefaultListSelectionModel.java:246)
	at java.desktop/javax.swing.DefaultListSelectionModel.insertIndexInterval(DefaultListSelectionModel.java:670)
	at java.desktop/javax.swing.plaf.basic.BasicListUI$Handler.intervalAdded(BasicListUI.java:2743)
	at java.desktop/javax.swing.AbstractListModel.fireIntervalAdded(AbstractListModel.java:153)
	at java.desktop/javax.swing.DefaultListModel.addElement(DefaultListModel.java:361)
	at org.mars_sim.msp.ui.swing.tool.search.SearchWindow$UnitListModel.updateList(SearchWindow.java:534)
	at org.mars_sim.msp.ui.swing.tool.search.SearchWindow$UnitListModel.unitManagerUpdate(SearchWindow.java:545)
	at org.mars_sim.msp.core.UnitManager.fireUnitManagerUpdate(UnitManager.java:2097)
	at org.mars_sim.msp.core.UnitManager.addUnit(UnitManager.java:582)
	at org.mars_sim.msp.core.Unit.<init>(Unit.java:197)
	at org.mars_sim.msp.core.equipment.Equipment.<init>(Equipment.java:70)
	at org.mars_sim.msp.core.equipment.Barrel.<init>(Barrel.java:39)
	at org.mars_sim.msp.core.equipment.EquipmentFactory.createEquipment(EquipmentFactory.java:86)
	at org.mars_sim.msp.core.equipment.EquipmentFactory.createEquipment(EquipmentFactory.java:76)
	at org.mars_sim.msp.core.equipment.EquipmentFactory.createEquipment(EquipmentFactory.java:112)
	at org.mars_sim.msp.core.equipment.ContainerUtil.getContainerPhase(ContainerUtil.java:164)
	at org.mars_sim.msp.core.structure.goods.GoodsManager.determineEquipmentDemand(GoodsManager.java:2935)
	at org.mars_sim.msp.core.structure.goods.GoodsManager.determineEquipmentGoodValue(GoodsManager.java:2896)
	at org.mars_sim.msp.core.structure.goods.GoodsManager.determineGoodValue(GoodsManager.java:608)
	at org.mars_sim.msp.core.structure.goods.GoodsManager.updateGoodValue(GoodsManager.java:573)
	at org.mars_sim.msp.core.structure.goods.GoodsManager.updateGoodsValuePrice(GoodsManager.java:529)
	at org.mars_sim.msp.core.structure.goods.GoodsManager.timePassing(GoodsManager.java:434)
	at org.mars_sim.msp.core.structure.Settlement.timePassing(Settlement.java:1382)
	at org.mars_sim.msp.core.UnitManager$SettlementTask.call(UnitManager.java:2353)
	at org.mars_sim.msp.core.UnitManager$SettlementTask.call(UnitManager.java:1)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:830)
Caused by: com.alee.managers.style.StyleException: Component styling is not supported: com.alee.laf.list.WebListCellRenderer$UIResource[List.cellRenderer,0,0,0x0,invalid,alignmentX=0.0,alignmentY=0.0,border=com.alee.utils.laf.WebBorder@95b8775,flags=256,maximumSize=,minimumSize=,preferredSize=,defaultIcon=,disabledIcon=,horizontalAlignment=LEADING,horizontalTextPosition=TRAILING,iconTextGap=4,labelFor=,text=Debra Johnson,verticalAlignment=CENTER,verticalTextPosition=CENTER]
	at com.alee.managers.style.StyleManager.getData(StyleManager.java:1358)
	at com.alee.managers.style.StyleManager.getSkin(StyleManager.java:1234)
	at com.alee.painter.decoration.AbstractDecorationPainter.collectDecorationStates(AbstractDecorationPainter.java:887)
	at com.alee.painter.decoration.AbstractDecorationPainter.updateDecorationState(AbstractDecorationPainter.java:1174)
	at com.alee.painter.decoration.AbstractDecorationPainter.propertyChanged(AbstractDecorationPainter.java:199)
	at com.alee.painter.AbstractPainter$1.propertyChange(AbstractPainter.java:442)
	at java.desktop/java.beans.PropertyChangeSupport.fire(PropertyChangeSupport.java:341)
	at java.desktop/java.beans.PropertyChangeSupport.firePropertyChange(PropertyChangeSupport.java:333)
	at java.desktop/java.beans.PropertyChangeSupport.firePropertyChange(PropertyChangeSupport.java:266)
	at java.desktop/java.awt.Component.firePropertyChange(Component.java:8732)
	at com.alee.laf.list.WebListCellRenderer.allowPropertyChange(WebListCellRenderer.java:410)
	at com.alee.laf.list.WebListCellRenderer.checkPropertyChange(WebListCellRenderer.java:392)
	at com.alee.laf.list.WebListCellRenderer.firePropertyChange(WebListCellRenderer.java:416)
	... 41 more
00-Adir-01:010.021 (Info) LivingAccommodations : [Kyocera - Lander Hab 1] Yua Kato - Designated a bed at (1.5, 3.16).
00-Adir-01:010.021 (Info) LivingAccommodations : [Kyocera - Lander Hab 1] Shota Fujita - Designated a bed at (-0.39, 3.46).
00-Adir-01:010.021 (Info) LivingAccommodations : [Kyocera - Lander Hab 1] Kosei Ishikawa - Designated a bed at (-3.14, 0.83).
00-Adir-01:010.021 (Info) LivingAccommodations : [Kyocera - Lander Hab 1] Mao Yamashita - Designated a bed at (-3.04, -1.04).
00-Adir-01:010.021 (Info) LivingAccommodations : [Kyocera - EVA Airlock 1] Airi Nakayama - Designated a bed at (4.9, 1.35).
00-Adir-01:010.021 (Info) LivingAccommodations : [Kyocera - EVA Airlock 1] Taisei Endo - Designated a bed at (4.9, -1.35).
```

**Comment by mokun** [comment]
btw, if I don't rush to open the Mission Tool and let the simulation runs for 10 seconds or so first, then it seems it's much less likely to happen.

**Comment by bevans2000** [comment]
There is another similar condition that I have found previously. If a new Mission is created when the UI is building then there is an exception in a TableModel. It seems all the windows are constructed when the UI starts instead of being created on demand when the user opens them. This accounts for the slow start-up of the UI.

**Comment by mokun** [comment]
> It seems all the windows are constructed when the UI starts instead of being created on demand when the user opens them. This accounts for the slow start-up of the UI.

There's always a tradeoff between the two approaches.

In the past, I recalled that as I clicked on a settlement row in the Monitor Tool in an attempt to open a `Settlement Window`, it takes forever. All those tabs were created on demand and thus causing a long delay for the `Settlement Window` to be constructed and popped up.

So I changed the behavior in such a way that it began to create each tab during the start-up so that the `Settlement Window` would be ready.

So, let's put it under the scope and reconsider what UI condition like these should change after v3.2.1 is out and optimizing it in v3.2.1, shall we ?
